### PR TITLE
feat: Provide documentation on various modes for injecting content into an ARM template

### PR DESCRIPTION
**Pull request by pim-simons**
Added documentation on the injection instructions into an ARM template and what they do.

Closes https://github.com/arcus-azure/arcus.scripting/issues/226

### CODE REVIEW

File: docs/preview/02-Features/powershell/arm.md [hunk 1]
```diff
@@ -54,5 +54,23 @@ Injecting the content:
...
 
-**Recommendations**
+### Injection Instructions
+It is possible to supply injection instructions in the injection annotation, this allows you to add specific functionality to the injection. These are the available injection instructions:
+
+| Injection Instruction | Description                                                                                                 |
+| --------------------- | ----------------------------------------------------------------------------------------------------------- |
+| `EscapeJson`          | Replace double quotes not preceded by a backslash with escaped quotes                                       |
+| `ReplaceSpecialChars` | Replace newline characters with literal equivalents, tabs with spaces and `"` with `\"`                     |
+| `InjectAsJsonObject`  | Tests if the content is valid JSON and makes sure the content is injected without surrounding double quotes |
```
Comment (stijnmoreels) [inline docs/preview/02-Features/powershell/arm.md:64]: Maybe we can add this update to the already released version. We could look if these things are already released, and update those releases too, right?

**Review by stijnmoreels — APPROVED** [review]
Great stuff! Valuable update, Pim. Was already 'lingering' on our issue list. Great initiative.

File: docs/preview/02-Features/powershell/arm.md (hunk 1 above)
Comment (pim-simons) [reply docs/preview/02-Features/powershell/arm.md:64]: Since it is not a code change and just documentation I think it should be no problem to add this to the already release version. I haven't done that before though, so I would need some guidance on how to do that.

Comment (stijnmoreels) [reply docs/preview/02-Features/powershell/arm.md:64]: When releasing the feature docs (in any Arcus repo), we run `npm run docusaursus docs:version vx.x.x` (as described [here](https://github.com/arcus-azure/arcus.scripting/tree/master/docs#versioning)). This results in a specific version folder in the `/docs` folder in the root of the repo. So for (example) version `v1.1.0`, will result in a folder `/docs/versioned_docs/version-v1.1.0` where the same structure is available as in the `/docs/preview` folder. 

So, in short, the only difference between preview and published docs, is `/docs/preview` and `/docs/versioned_docs/version-vx.x.x` folders.

I think the ARM injection is already published for quite some time, so it may be available already very early. Looking in the tags of the repo (https://github.com/arcus-azure/arcus.scripting/tree/v0.4.3) can help.

Comment (pim-simons) [reply docs/preview/02-Features/powershell/arm.md:64]: I have included it in the previous versions of the documentation as well, see [`e7a388f` (#269)](https://github.com/arcus-azure/arcus.scripting/pull/269/commits/e7a388f3fbc12314b1332702487168ade2483864)

**Review by stijnmoreels — APPROVED** [review]
Perfect! Thx. 👌
